File: algorithms/0007/main.go
package p0007

import (
	"math"
)

func reverse(x int) int {
	sign := 1
	if x < 0 {
		x = -1 * x
		sign = -1
	}
	result := 0
	for x > 0 {
		tmp := x % 10
		result = result*10 + tmp
		x = x / 10
	}
	if result > math.MaxInt32 {
		return 0
	}
	return result * sign
}

File: algorithms/0007/main.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
  let sign
  if (x < 0) {
    sign = -1
    x = -x
  } else {
    sign = 1
  }
  let result = 0
  while (x > 0) {
    const tmp = x % 10
    result = result * 10 + tmp
    x = Math.floor(x / 10)
  }
  const MAX_INT = 2147483647
  return result > MAX_INT ? 0 : result * sign
}

module.exports = reverse
